### PR TITLE
feat: create user on first login if not exist 

### LINKED ISSUE #115: [auth] create/read user object on signup/login
## Features

- populate next-auth with relevant user data

### PR DESCRIPTION
**Pull request by davidlhw**
closes #115 

## context

currently, when the user signs up, the user object is created on supabase’s auth table only. this user is what supabase uses to verify user authentication methods like `signIn` or `signUp`

however, the `public.user` table that the application have context to won't have this new user created on sign up

this is an issue because we cant embed data like user's university to the user payload that is used by nextauth to provide auth context. meaning, new users would be in "limbo" in the application POV

this PR's main goal is to also create the user on the application table when user signs up

## changes

- add logic to create new user on sign up
- add db table `university_domains` to track all domains belonging to the university
- make uni abbrv unique

## implementation

1. when user sign up, get user email
2. parse domain from email
3. filter for uni that this email domain belong to
4. create user based on defined fields (username is predefined with a random id)

### university abbreviation uniqueness

since the abbreviation is created and used in application layer, we can guarantee it's uniqueness, even if 2 university happen to actually have the same abbreviation of their name, we can include an added identifier

for example, in a situation like Singapore Management University vs Southern Methodist University, we can have `SMU` and `SMU-TX` as the abbreviation, where TX is for Texas, the state that Southern Methodist University is in. we dont necessarily have to follow this syntax, as long as the Abbrv makes some logical sense

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -73,10 +73,23 @@ model Universities {
...
 
+model UniversityDomains {
+    id                   Int          @id @default(autoincrement())
+    domain               String       @unique
+    belongToUniversity   Universities @relation(fields: [belongToUniversityId], references: [id])
+    belongToUniversityId Int          @map("belong_to_university")
+    createdAt            DateTime     @default(now()) @map("created_at") @db.Timestamptz(3)
+    updatedAt            DateTime     @updatedAt @map("updated_at") @db.Timestamptz(3)
+
+    @@unique([domain, belongToUniversityId])
```
Comment (Jaylin0312) [inline prisma/schema.prisma:89]: Question: What's the reason of creating a composite key? Do we expect 2 unis to share the same domain?

**Review by Jaylin0312 — DISMISSED** [review]
Overall looks good, left a comment as a question

File: prisma/schema.prisma (hunk 1 above)
Comment (davidlhw) [reply prisma/schema.prisma]: thanks for the review!

dont expect 2 unis to share same domain, but that check is already accounted for by declaring uniqueness for domains

removed this redundant composite unique keys at [8e81fcd](https://github.com/AfterClass-io/afterclass.io-v2/pull/136/commits/8e81fcd7ffa1450c1ce0919b23adfe6cee0bd640)

**Review by Jaylin0312 — APPROVED** [review]
clgtm at [9d48cad](https://github.com/AfterClass-io/afterclass.io-v2/pull/136/commits/9d48cad2af6f74077b8e4178365cc68c494ca835)

**Pull request merged**